### ISSUE TITLE
Error After Updating Nextflow and Phoenix Versions

### ISSUE THREAD

**Issue by Ashita2619** [CDCgov/phoenix#179]
Hello Everyone,

After updating both Nextflow and Phoenix to the latest versions, I am encountering an error while running the pipeline. The error occurs during the CDC_PHOENIX:PHOENIX_EXQC:ASSET_CHECK (1) process.

The pipeline fails with the error message mentioned below.
![image](https://github.com/user-attachments/assets/776bf2d7-8763-41ba-930f-55d8d9bd07fe)

1. I have verified the paths to the input files and dependencies.
2. I have checked the Nextflow and Phoenix logs, but I am unable to identify the root cause of the issue.

Question: 
1. Are there any known issues with the updated versions of Nextflow and Phoenix?
2. Could the error be related to changes in file paths or input/output configurations introduced in the new versions?

Thanks,
Ashita Jawali

**Comment by jvhagey** [comment]
Hi Ashita, what version of nextflow are you using? I don't know of an issue with particular versions of Nextflow and PHoeNIx except for some issues with Docker. I think the issue is that the output file name was changed from `Phoenix_Output_Report.tsv` to `Phoenix_Summary.tsv` in [v2.0.2](https://github.com/CDCgov/phoenix/blob/main/CHANGELOG.md#v202-08032023). I think if you just change that name reference in the wrapper you will be good. Let me know if that resolves the issue. We changed the name of the file in response to feedback from our compliance program. Any files with "report" in the name were replaced by "summary" in file names to avoid confusion regarding the difference between public health results (i.e. summary) and diagnostic results (i.e. report).

**Comment by Ashita2619** [comment]
The Nextflow version I am using currently is 24.10.3 and the Phoenix version is 2.1.1. I tried changing the Phoenix_Output_Report.tsv to Phoenix_Summary.tsv as per your suggestion in my helper function but still I am getting the same error.

**Comment by jvhagey** [comment]
Hey, so I ran a set of samples with the same version of nextflow and phoenix and didn't get any error so I think there is some other hiccup coming from the wrapper script. Is the author of the script, Adrian I assume (?), still around to look into it? If not I can try to pull the script and see what the issue is.

**Comment by Ashita2619** [comment]
No, Adrian is not around to look into it he left the organization. It would be of great help if you could look into the script. 

Thanks,
Ashita

**Comment by jvhagey** [comment]
ok, after pulling the script and running it with singularity it finished fine. So that got me thinking this was a docker issue and looks like it was brought up in another issue already https://github.com/CDCgov/phoenix/issues/175 . Can you implement the change suggested there and try again?

**Comment by Ashita2619** [comment]
I implemented the changes as mentioned and it was working fine until i got this error which is attached below:

![image](https://github.com/user-attachments/assets/28ca0ca1-b7a2-4093-be80-a02dd9904041)

**Comment by jvhagey** [comment]
So this is related to the kraken2 database. Check the path you are giving it, it should have the file `ktaxonomy.tsv` in the krakendb folder you give it. You should be using a kraken2db from [here](https://benlangmead.github.io/aws-indexes/k2), I suggest the standard-8 version. you want to click on the `.tar.gz` link to download it. As of March 2023, each index includes the required `ktaxonomy.tsv` file.

**Comment by Ashita2619** [comment]
I am getting this one error which is at the final step to complete the pipeline I tried to pull the docker using this command  to get the newer image-  docker pull quay.io/jvhagey/phoenix  

![image](https://github.com/user-attachments/assets/6ef0fe18-74f7-4a6c-aded-f070a4d4b898)

**Comment by jvhagey** [comment]
Ok, I was trying to install the same conda environment you are running nextflow in to check it, but it keeps stalling out on my system. This looks like nextflow needs `ps`, can you try running the command `ps` in your nextflow environment? I assume its not there. If its not there then install it with `conda install conda-forge::procps-ng` in your activate conda environment. I will add `ps` to the base container in its next version. let me know if that fixes it.

**Comment by Ashita2619** [comment]
So, I tried running the command ps --version just to check if its present in the conda environment where Nextflow is and it turns out that I have it installed but still I am getting the same error.

**Comment by jvhagey** [comment]
ok, I am going to make a new container and will let you know when its posted.

**Comment by NoForkJustCommit** [comment]
Sorry for the delay, so when checked the container has `ps` in it. And you wouldn't actually be pulling the containers if you are running it in a conda environment, so us making a new one doesn't really help. What is the command(s) you are using to run it? or is there a bash script or something on the repo you can point me to? I know this is going slow and we are doing our best.

**Comment by Ashita2619** [comment]
I am running this command :-
subprocess.run(". $CONDA_PREFIX/home/ssh_user/mambaforge/etc/profile.d/conda.sh && conda activate nextflow && nextflow run "+path_to_phoenix+"/main.nf -profile docker -entry CDC_PHOENIX --input "+phoniex_samplesheet+" --kraken2db "+path_to_kraken+" --outdir "+output_dir,shell=True)                                                       
  
which is there in the https://github.com/Ashita2619/CRAB_Analysis/blob/main/scripts/Phoenix_Launcher/WF_1_helper.py script. Let me know if you need anything else from me.

Thanks!

**Comment by Jon-gerhart** [comment]
Hello,
I believe the issue is related to the shell environment spawned from subprocess.open(), there also appears to be an inaccuracy in this line's second argument as "$CONDA_PREFIX/home/ssh_user/..." should not resolve.

Can you please try amending line 36 of "WF_1_helper.py" from 

subprocess.run(". $CONDA_PREFIX/home/ssh_user/mambaforge/etc/profile.d/conda.sh && conda activate nextflow && nextflow run "+path_to_phoenix+"/main.nf -profile docker -entry CDC_PHOENIX --input "+phoniex_samplesheet+" --kraken2db "+path_to_kraken+" --outdir "+output_dir,shell=True)
 
to

subprocess.run("source $HOME/.bashrc && source /home/ssh_user/mambaforge/etc/profile.d/conda.sh && conda activate nextflow && nextflow run "+path_to_phoenix+"/main.nf -profile docker -entry CDC_PHOENIX --input "+phoniex_samplesheet+" --kraken2db "+path_to_kraken+" --outdir "+output_dir , shell=True)

And let us know if this resolves the issue?

**Comment by Jon-gerhart** [comment]
Hello @Ashita2619 
Just following up on this issue, did the suggested line amendment to the subprocess command work for you on your system?
If you are still experiencing this issue let me know so I know to investigate it further.

**Comment by Ashita2619** [comment]
Hello @Jon-gerhart 

I was able to run the workflow by changing the line you suggested to the subprocess command. 

Thanks for your support.
~ Ashita Jawali
